lib/maplibre-compose/src/commonMain/kotlin/dev/sargunv/maplibrecompose/core/expression/ExpressionScope.kt

```diff
@@ -82,18 +83,23 @@ public object ExpressionScope {
 
   // region Variable binding
 
+  @JvmInline
+  public value class Variable<@Suppress("unused") T : ExpressionValue>(public val name: String)
+
+  /** Declares a named variable for use in [bind] and [use]. */
+  public fun <T : ExpressionValue> declare(name: String): Variable<T> = Variable(name)
+
   /**
-   * Binds expressions to named variables, which can then be referenced in the result expression
-   * using [useVariable].
+   * Binds expressions to variables declared by [declare], which can then be referenced with [use]
+   * in the [expression].
    */
-  public fun <T : ExpressionValue> withVariable(
-    name: String,
-    value: Expression<*>,
+  public fun <T : ExpressionValue> Variable<T>.bind(
```

Comment:
This is cool, on the one hand, on the other hand, the expressiveness with "withVariable" is lost: It is a really odd construct to have the rest of the expression be part of the declaration/binding of a variable, "withVariable" explained what's happening a bit. `variable.bind` doesn't. Also, separating the declaration and binding does give room for an additional misunderstanding/mistake  that couldn't happen if both were one: That a user declares a variable and then tries to use it, without "binding" it.

Reply:
The same misunderstanding could happen if the variable is a string 

I might have an idea to deal with this, will hack on it in a bit

Reply:
Something like:

```kt
withVariable(const(1)) { var ->
  var.value()
}
```

Only question there is how we get the string name. It's never shown to the user, so maybe auto generated? Or an optional parameter to withVariable

Reply:
hmm, I don't know. The scope should contain and *only* contain an expression, right? What happens if there is more?

Reply:
it should return an expression. I imagine it'd be a call-in-place inlined block (kind of like `let`) so adding more code there is the same as adding more code right next to the expression

Reply:
it's now part of https://github.com/sargunv/maplibre-compose/pull/133/files

```kt
val example: Expression<StringValue> = withVariable("foo", const(1)) {
  foo -> foo.use().convertToString()
}
```